src/site/content/en/react/how-amp-can-guarantee-fastness-in-your-nextjs-app/index.md

```diff
@@ -0,0 +1,99 @@
+
+* **AMP HTML**: A restricted form of HTML that makes certain optimizations mandatory and prohibits
+  architectural patterns that lead to slowness. See [How AMP works][how] for a high-level
+  overview of the optimizations and restrictions.
+* **AMP Cache**: A content cache used by some search engines, such as Google and Bing, that uses
+  [prerendering] to speed up page loads. See [Why AMP Caches exist][why] to learn more about the
+  benefits and tradeoffs of the caches and [How does my AMP page get cached?][how] to understand
+  how your AMP pages get into the caches.
+
+## How you can use AMP in Next.js {: #strategies }
+
+There are two ways to use AMP in Next.js:
+
+* The **Hybrid AMP** approach lets you create an accompanying AMP version of any
+  Next.js page.
+* The **AMP-only** approach lets you make AMP the only option for a page.
+
+Although Next.js is usually thought of as a React framework, it's important to understand that
+when you use Next.js to serve AMP pages, you can no longer run React components client-side because
+React components are not valid AMP components. In other words, Next.js is no longer a React
+framework but rather a server-side templating engine for generating AMP pages.
+
+{% Aside 'codelab' %}
+  Check out [How to use AMP in Next.js](/how-to-use-amp-in-nextjs) to try out the two
+  approaches yourself.
+{% endAside %}
+
+## How to decide whether to use the Hybrid AMP or AMP-only approach {: #guidance }
+
+If you're serious about load performance, an AMP-only page could be a good way to make sure
+that your page gets fast and stays fast. But here's the catch: in order to guarantee fastness,
+AMP must prohibit certain architectural patterns and HTML elements that often lead to slow pages.
+For example, AMP doesn't allow custom synchronous JavaScript because 
+[render-blocking resources][blockers] are a common cause of slow page loads.
+
+In order to understand whether an AMP-only approach is right for you, you need to figure out
+whether all of your frontend code can be represented in AMP HTML:
+
+* Read [How AMP works](https://amp.dev/about/how-amp-works/) to understand AMP's high-level
+  architectural restrictions and mandatory optimizations.
+* Read [HTML Tags][tags] to see what HTML tags AMP allows and prohibits,
+  browse the [AMP component catalogue](https://amp.dev/documentation/components/) to see the
+  custom components that the AMP community has built to solve common use cases, and check
+  out [amp-script] to learn how to use custom JavaScript to implement features that AMP
+  doesn't currently support.
+
+Even if an AMP-only approach won't work for your page, it might still be a good idea to
+use AMP whenever possible, because of the guaranteed fastness of AMP HTML and the AMP Caches.
+The Hybrid AMP approach in Next.js provides a way to conditionally serve AMP pages. However,
+it also creates a higher maintenance cost because it requires you to maintain
+two versions of each page.
```

Comment:
I was going to suggest "it also creates" -> "can result in", but I think this is fine after all. Why: Some people generate both from a shared code base, converting one to the other. So yes, you need to maintain a conversion tool, but they don't have to maintain two code bases for every page. That was why I was a little hesitant to be so definitive about having to maintain two pages. In Next.js it is somewhat true (that is how they address the problem), so I decided current wording is okay.